components/experimental/src/units/ratio.rs

```diff
@@ -0,0 +1,180 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use core::{
+    ops::{Add, AddAssign, Div, DivAssign, Mul, MulAssign, Sub, SubAssign},
+    str::FromStr,
+};
+
+use num_bigint::BigInt;
+use num_rational::Ratio;
+use num_traits::Signed;
+use num_traits::ToPrimitive;
+use num_traits::{One, Pow, Zero};
+
+// TODO: implement AsULE for IcuRatio and use it in Data module.
+// TODO: add test cases for IcuRatio.
+/// A ratio type that uses `BigInt` as the underlying type.
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct IcuRatio(Ratio<BigInt>);
+
+#[derive(Debug)]
+pub enum IcuRatioError {
+    BigIntParseError(num_bigint::ParseBigIntError),
+    InvalidRatioString,
+}
+
+impl IcuRatio {
+    pub fn new(numerator: BigInt, denominator: BigInt) -> Self {
+        Self(Ratio::new(numerator, denominator))
+    }
+
+    pub fn numerator(&self) -> &BigInt {
+        self.0.numer()
+    }
+
+    pub fn numerator_str(&self) -> String {
+        self.0.numer().to_string()
+    }
+
+    pub fn denominator(&self) -> &BigInt {
+        self.0.denom()
+    }
+
+    pub fn denominator_str(&self) -> String {
+        self.0.denom().to_string()
+    }
+
+    /// Returns the reciprocal of the ratio.
+    /// For example, the reciprocal of 2/3 is 3/2.
+    pub fn recip(&self) -> Self {
+        Self(self.0.recip())
+    }
+
+    /// Returns the absolute value of the ratio.
+    pub fn abs(&self) -> Self {
+        Self(self.0.abs())
+    }
+
+    pub fn ten() -> Self {
+        Self(Ratio::from_integer(10.into()))
+    }
+
+    pub fn two() -> Self {
+        Self(Ratio::from_integer(2.into()))
+    }
+
+    pub fn to_f64(&self) -> Option<f64> {
+        self.0.to_f64()
+    }
+}
```

Comment:
Suggestion: please implement `apply_si_prefix` as suggested in https://github.com/unicode-org/icu4x/pull/4404#discussion_r1467064609 and `From<ConversionInfoULE> for IcuRatio` as suggested in https://github.com/unicode-org/icu4x/pull/4404#discussion_r1474970310

Reply:
You can:

1. `impl From<&ConversionInfoULE>` 
3. `impl From<ConversionInfo>`
2. Make a constructor `pub fn from_conversion_info(&ConversionInfoULE)`

Reply:
I implement `factor_from...` and `offset_from...`